dptb/nnops/loss.py

```
@@ -643,6 +643,95 @@ def forward(self, data: AtomicDataDict, ref_data: AtomicDataDict):
             return 0.5 * (onsite_loss + hopping_loss)
 
 
+@Loss.register("hamil_abs_mae")
+class HamilLossAbsMAE(nn.Module):
+    def __init__(
+            self,
+            basis: Dict[str, Union[str, list]] = None,
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)